### PR TITLE
HIVE-19815: Repl dump should not propagate the checkpoint and repl source properties.

### PR DESCRIPTION
**Pull request by sankarh**

### CODE REVIEW

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcrossInstances.java [hunk 1]
```diff
@@ -811,4 +819,120 @@ public void testIfCkptSetForObjectsByBootstrapReplLoad() throws Throwable {
...
+            .run("use " + primaryDbName)
+            .run("create table t1 (place string) partitioned by (country string)")
+            .run("insert into table t1 partition(country='india') values ('bangalore')")
+            .dump(primaryDbName, null);
+
+    // Bootstrap Repl A -> B
+    WarehouseInstance.Tuple tupleReplica = replica.load(replicatedDbName, tuplePrimary.dumpLocation)
+            .run("repl status " + replicatedDbName)
+            .verifyResult(tuplePrimary.lastReplicationId)
+            .run("alter database " + replicatedDbName + " set dbproperties ('" + SOURCE_OF_REPLICATION + "' = '1, 2, 3')")
```
Comment (maheshk114) [inline itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcrossInstances.java]: can do a dump before setting the property to check the property is not propagated ..dump should fail

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcrossInstances.java [hunk 2]
```diff
@@ -811,4 +819,120 @@ public void testIfCkptSetForObjectsByBootstrapReplLoad() throws Throwable {
...
+            .run("repl status " + replicatedDbName)
+            .verifyResult(tuplePrimaryInc.lastReplicationId)
+            .dump(replicatedDbName, tupleReplica.lastReplicationId);
+
+    // Check if DB in B have ckpt property is set to bootstrap dump location used in B and missing for table/partition.
+    db = replica.getDatabase(replicatedDbName);
+    verifyIfCkptSet(db.getParameters(), tuplePrimary.dumpLocation);
+    t1 = replica.getTable(replicatedDbName, "t1");
+    verifyIfCkptPropMissing(t1.getParameters());
+    india = replica.getPartition(replicatedDbName, "t1", Collections.singletonList("india"));
```
Comment (maheshk114) [inline itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcrossInstances.java]: can check other properties in the table are not removed

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcrossInstances.java [hunk 3]
```diff
@@ -811,4 +819,120 @@ public void testIfCkptSetForObjectsByBootstrapReplLoad() throws Throwable {
...
+    replica.load(replDbFromReplica, tupleReplicaInc.dumpLocation)
+            .run("use " + replDbFromReplica)
+            .run("repl status " + replDbFromReplica)
+            .verifyResult(tupleReplicaInc.lastReplicationId);
+
+    // Check if DB/table/partition in C doesn't have repl.source.for props. Also ensure, ckpt property
+    // in DB is set to bootstrap dump location used in C but for table/partition, it is missing.
+    db = replica.getDatabase(replDbFromReplica);
+    verifyIfCkptSet(db.getParameters(), tupleReplica.dumpLocation);
+    verifyIfSrcOfReplPropMissing(db.getParameters());
```
Comment (maheshk114) [inline itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcrossInstances.java]: i think there is a inconsistency in the behavior ..for database the ckpt property is not removed after incremental ..but for table and partition it is removed.

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcrossInstances.java [hunk 4]
```diff
@@ -811,4 +819,120 @@ public void testIfCkptSetForObjectsByBootstrapReplLoad() throws Throwable {
     Partition uk = replica.getPartition(replicatedDbName, "t2", Collections.singletonList("uk"));
     verifyIfCkptSet(uk.getParameters(), tuple.dumpLocation);
   }
+
+  @Test
+  public void testIfCkptAndSourceOfReplPropsIgnoredByReplDump() throws Throwable {
+    WarehouseInstance.Tuple tuplePrimary = primary
+            .run("use " + primaryDbName)
```
Comment (maheshk114) [inline itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcrossInstances.java]: can add a test case, database created with source for replication property and then dump and load (to check create database handler)

File: ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/io/TableSerializer.java [hunk 5]
```diff
@@ -65,7 +68,16 @@ public void writeTo(JsonWriter writer, ReplicationSpec additionalPropertiesProvi
     }
   }
 
-  private Table addPropertiesToTable(Table table, ReplicationSpec additionalPropertiesProvider) {
+  private Table updatePropertiesInTable(Table table, ReplicationSpec additionalPropertiesProvider) {
+    // Remove all the entries from the parameters which are added by repl tasks internally.
+    Map<String, String> parameters = table.getParameters();
+    if (parameters != null) {
+      Map<String, String> tmpParameters = new HashMap<>(parameters);
```
[the remark this answers is not in the document]
Comment (sankarh) [reply ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/io/TableSerializer.java]: I think yes, copy pasted from somewhere. will remove directly.

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcrossInstances.java (hunk 1 above)
Comment (sankarh) [reply itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcrossInstances.java]: This case is not in the scope of this ticket. It was already tested by TestReplicationScenarios. testDumpNonReplDatabase test case.

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcrossInstances.java (hunk 2 above)
Comment (sankarh) [reply itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcrossInstances.java]: ok.

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcrossInstances.java (hunk 3 above)
Comment (sankarh) [reply itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcrossInstances.java]: That's due to different behaviour for AlterDatabase and AlterTable in base code. However, this is not an issue as it doesn't affect state of target DB.

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcrossInstances.java (hunk 4 above)
Comment (sankarh) [reply itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcrossInstances.java]: We cannot test Create Database handler as bootstrap doesn't use it. Also, cannot be part of incremental as well.

File: ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/io/TableSerializer.java [hunk 6]
```diff
@@ -18,9 +18,11 @@
 package org.apache.hadoop.hive.ql.parse.repl.dump.io;
 
 import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.ReplChangeManager;
```
[the remark this answers is not in the document]
Comment (sankarh) [reply ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/io/TableSerializer.java]: ok

**Pull request closed without merging**